flow/connectors/core.go

```diff
@@ -136,11 +136,15 @@ type CDCNormalizeConnector interface {
 type QRepPullConnector interface {
 	Connector
 
+	// GetTableSchema returns the schema of a table.
+	GetTableSchema(ctx context.Context, req *protos.GetTableSchemaBatchInput) (*protos.GetTableSchemaBatchOutput, error)
+
 	// GetQRepPartitions returns the partitions for a given table that haven't been synced yet.
 	GetQRepPartitions(ctx context.Context, config *protos.QRepConfig, last *protos.QRepPartition) ([]*protos.QRepPartition, error)
 
 	// PullQRepRecords returns the records for a given partition.
-	PullQRepRecords(ctx context.Context, config *protos.QRepConfig, partition *protos.QRepPartition) (*model.QRecordBatch, error)
+	PullQRepRecordStream(ctx context.Context, config *protos.QRepConfig,
```

Comment:
can create something like a QRepPullStreamConnector interface instead, prefer that, fallback to QRepPullConnector when ErrUnsupportedFunctionality